dbt/include/bigquery/macros/catalog.sql

```diff
@@ -7,30 +7,42 @@
   {%- else -%}
 
   {%- set query -%}
-    with tables as (
+    with materialized_views as (
         select
-            project_id as table_database,
-            dataset_id as table_schema,
-            table_id as original_table_name,
+            table_catalog as project_id,
+            table_schema as dataset_id,
+            table_name as table_id
+        from {{ information_schema.replace(information_schema_view='MATERIALIZED_VIEWS') }}
+    ),
+    tables as (
+        select
+            tables.project_id as table_database,
+            tables.dataset_id as table_schema,
+            tables.table_id as original_table_name,
 
-            concat(project_id, '.', dataset_id, '.', table_id) as relation_id,
+            concat(tables.project_id, '.', tables.dataset_id, '.', tables.table_id) as relation_id,
 
-            row_count,
-            size_bytes as size_bytes,
+            tables.row_count,
+            tables.size_bytes as size_bytes,
             case
-                when type = 1 then 'table'
-                when type = 2 then 'view'
+                when materialized_views.table_id is not null then 'materialized view'
+                when tables.type = 1 then 'table'
+                when tables.type = 2 then 'view'
                 else 'external'
             end as table_type,
 
-            REGEXP_CONTAINS(table_id, '^.+[0-9]{8}$') and coalesce(type, 0) = 1 as is_date_shard,
-            REGEXP_EXTRACT(table_id, '^(.+)[0-9]{8}$') as shard_base_name,
-            REGEXP_EXTRACT(table_id, '^.+([0-9]{8})$') as shard_name
+            REGEXP_CONTAINS(tables.table_id, '^.+[0-9]{8}$') and coalesce(type, 0) = 1 as is_date_shard,
```

Comment:
These regexes make me uncomfortable. Is there any way to pull them out as vars or something so we can at least document them a la C macros? If nothing else, it might be nice to know those that should be the same, do in fact stay the same

Reply:
They were already there. Those lines show up because I suffixed the column name with `tables.`. Perhaps we should revisit them (I share your concern), but that would be outside of the scope of this PR.

Reply:
I can get behind that. They are clearing working right now. I just find the whole thing uncomfy as coding practice. However, since it's just a simple/intentional copy-paste job, I can get behind that.